### PR TITLE
added validation framework

### PR DESCRIPTION
**Pull request by ShubhranshuSanjeev**
The library now includes a validation module to ensure JSON Logic rules conform to your requirements

### CODE REVIEW

File: src/validation.rs [hunk 1]
```diff
@@ -0,0 +1,95 @@
...
+#[derive(Debug, Clone, Copy)]
+pub struct RequireAndWrapper {
+    /// If true, allows empty JSON logic objects (e.g., `{}`)
+    pub allow_empty: bool,
+}
+
+impl Default for ValidationConfig {
+    fn default() -> Self {
+        ValidationConfig {
+            // allowed_operators: None,
```
Comment (Datron) [inline src/validation.rs:58]: In the future can we have allowed_logical_operators and allowed_boolean_operators

File: src/validation.rs [hunk 2]
```diff
@@ -0,0 +1,95 @@
...
+            return Err(ValidationError::new("Empty JSON Logic is not allowed", "$"));
+        }
+    }
+
+    // Check if the root operator is 'and'
+    if let Some(obj) = json_logic.as_object() {
+        if obj.len() == 1 && obj.contains_key("and") {
+            return Ok(());
+        }
+    }
```
Comment (Datron) [inline src/validation.rs]: @ShubhranshuSanjeev I recommend this be a config, incase some other team in juspay wants to use jsonlogic-rs and they don't want this behaviour

**Review by Datron — APPROVED** [review]

File: src/validation.rs (hunk 2 above)
Comment (ShubhranshuSanjeev) [reply src/validation.rs]: @Datron , its already a config. Not sure what you mean, could you be please elaborate?

File: src/validation.rs [hunk 3]
```diff
@@ -0,0 +1,95 @@
...
+    /// Empty JSON logic objects are allowed if allow_empty is true.
+    pub require_and_wrapper: Option<RequireAndWrapper>,
+}
+
+/// Configuration for requiring JSON Logic to be wrapped in an 'and' block
+#[derive(Debug, Clone, Copy)]
+pub struct RequireAndWrapper {
+    /// If true, allows empty JSON logic objects (e.g., `{}`)
+    pub allow_empty: bool,
+}
```
Comment (Datron) [inline src/validation.rs:53]: Do we need a full struct for this?

File: src/validation.rs [hunk 4]
```diff
@@ -0,0 +1,95 @@
...
+
+impl Default for ValidationConfig {
+    fn default() -> Self {
+        ValidationConfig {
+            // allowed_operators: None,
+            // max_depth: None,
+            // required_variables: None,
+            // allowed_variables: None,
+            // data_schema_validator: None,
+            require_and_wrapper: None,
```
Comment (Datron) [inline src/validation.rs:63]: Can this just be true/false? Or Option<bool>?

File: src/validation.rs (hunk 2 above)
Comment (Datron) [reply src/validation.rs]: Got it, read this wrong. Nevermind

File: README.md [hunk 5]
```diff
@@ -82,3 +82,40 @@ For Rust usage examples and edge cases have a look at the linked tests for each
     - [`jp_ver_ge`](https://github.com/juspay/jsonlogic_rs/blob/master/tests/jp_version.rs#L50)
     - [`jp_ver_lt`](https://github.com/juspay/jsonlogic_rs/blob/master/tests/jp_version.rs#L73)
     - [`jp_ver_le`](https://github.com/juspay/jsonlogic_rs/blob/master/tests/jp_version.rs#L96)
+
+## Validation
+
+The library now includes a validation module to ensure JSON Logic rules conform to your requirements:
```
Comment (ayushjain17) [inline README.md:88]:
```suggestion
The library now includes a validation module to ensure JSON Logic rules conform to your requirements and restrictions:
```

File: README.md [hunk 6]
```diff
@@ -82,3 +82,40 @@ For Rust usage examples and edge cases have a look at the linked tests for each
...
+});
+
+match validate(&rule, &config) {
+    Ok(_) => println!("Rule is valid!"),
+    Err(err) => println!("Validation failed: {} at {}", err.message, err.path),
+}
+```
+
+The validation module lets you:
+- Require rules to be wrapped in an 'and' block (with option to allow empty rules)
```
Comment (ayushjain17) [inline README.md:114]: this is very restrictive and specific to our use case only, ideally we should be able to generalise this further

File: src/validation.rs [hunk 7]
```diff
@@ -0,0 +1,95 @@
...
+/// Validates that JSON Logic is properly wrapped in an 'and' block.
+fn validate_and_wrapper(json_logic: &Value, config: &RequireAndWrapper) -> ValidationResult {
+    // Allow empty objects if configured
+    if json_logic.is_object() && json_logic.as_object().unwrap().is_empty() {
+        if config.allow_empty {
+            return Ok(());
+        } else {
+            return Err(ValidationError::new("Empty JSON Logic is not allowed", "$"));
+        }
+    }
```
Comment (ayushjain17) [inline src/validation.rs]: can we please avoid using any `serde_json` operations here as they are not safe and do not entirely cover everything

Rather please get the AST first using the already existing functions and perform checks on that

**Review by ayushjain17 — CHANGES_REQUESTED** [review]
need to re-review once post the changes

**Review by ayushjain17 — APPROVED** [review]
